src/plugins/credential_management/public/service/creation/manager.ts

```diff
@@ -10,61 +10,59 @@
  */
 
 import { HttpSetup } from '../../../../../core/public';
-// import { CredentialCreationConfig, UrlHandler, CredentialCreationOption } from './config';
+import { CredentialCreationConfig, UrlHandler, CredentialCreationOption } from './config';
 
 // TODO: Revisit it
 export class CredentialCreationManager {
-//   private configs: CredentialCreationConfig[] = [];
+  private configs: CredentialCreationConfig[] = [];
   setup(httpClient: HttpSetup) {
     return {
-    // TODO - Add Config  
-    //   addCreationConfig: (Config: typeof CredentialCreationConfig) => {
-    //     const config = new Config({ httpClient });
+      addCreationConfig: (Config: typeof CredentialCreationConfig) => {
+        const config = new Config({ httpClient });
 
-    //     if (this.configs.findIndex((c) => c.key === config.key) !== -1) {
-    //       throw new Error(`${config.key} exists in CredentialCreationManager.`);
-    //     }
+        if (this.configs.findIndex((c) => c.key === config.key) !== -1) {
+          throw new Error(`${config.key} exists in CredentialCreationManager.`);
+        }
 
-    //     this.configs.push(config);
-    //   },
+        this.configs.push(config);
+      },
     };
   }
 
   start() {
-    // const getType = (key: string | undefined): CredentialCreationConfig => {
-    //   if (key) {
-    //     const index = this.configs.findIndex((config) => config.key === key);
-    //     const config = this.configs[index];
+    const getType = (key: string | undefined): CredentialCreationConfig => {
+      if (key) {
+        const index = this.configs.findIndex((config) => config.key === key);
+        const config = this.configs[index];
 
-    //     if (config) {
-    //       return config;
-    //     } else {
-    //       throw new Error(`Index pattern creation type not found: ${key}`);
-    //     }
-    //   } else {
-    //     return getType('default');
-    //   }
-    // };
+        if (config) {
+          return config;
+        } else {
+          throw new Error(`Credential creation type not found: ${key}`);
+        }
+      } else {
+        return getType('default');
```

Comment:
what if on entry for `default` will this cause infinity recursive call?

Reply:
removed the getType logic